Source/Core/VideoCommon/PerformanceMetrics.cpp

```diff
@@ -0,0 +1,254 @@
+
+  // Change Color based on % Speed
+  float r = 0.0f, g = 1.0f, b = 1.0f;
+  if (g_ActiveConfig.bShowSpeedColors)
+  {
+    r = 1.0 - (speed - 80.0) / 20.0;
+    g = speed / 80.0;
+    b = (speed - 90.0) / 10.0;
+  }
+
+  const float window_padding = 8.f * backbuffer_scale;
+  const float window_width = 93.f * backbuffer_scale;
+  float window_y = window_padding;
+  float window_x = ImGui::GetIO().DisplaySize.x - window_padding;
+
+  const float graph_width = 50.f * backbuffer_scale + 3.f * window_width + 2.f * window_padding;
+  const float graph_height =
+      std::min(200.f * backbuffer_scale, ImGui::GetIO().DisplaySize.y - 85.f * backbuffer_scale);
+
+  ImGui::PushStyleVar(ImGuiStyleVar_WindowBorderSize, 0.f);
+  ImGui::PushStyleVar(ImGuiStyleVar_WindowRounding, 14.f * backbuffer_scale);
+  if (g_ActiveConfig.bShowGraphs)
+  {
+    ImGui::PushStyleVar(ImGuiStyleVar_WindowPadding, ImVec2(0, 4.f * backbuffer_scale));
+
+    // Position in the top-right corner of the screen.
+    ImGui::SetNextWindowPos(ImVec2(window_x, window_y), ImGuiCond_Always, ImVec2(1.0f, 0.0f));
+    ImGui::SetNextWindowSize(ImVec2(graph_width, graph_height));
+    ImGui::SetNextWindowBgAlpha(bg_alpha);
+    window_y += graph_height + window_padding;
+
+    if (ImGui::Begin("PerformanceGraphs", nullptr, imgui_flags))
+    {
+      const static int num_ticks = 17;
+      const static double tick_marks[num_ticks] = {0.0,
+                                                   1000.0 / 360.0,
+                                                   1000.0 / 240.0,
+                                                   1000.0 / 180.0,
+                                                   1000.0 / 120.0,
+                                                   1000.0 / 90.00,
+                                                   1000.0 / 59.94,
+                                                   1000.0 / 40.00,
+                                                   1000.0 / 29.97,
+                                                   1000.0 / 24.00,
+                                                   1000.0 / 20.00,
+                                                   1000.0 / 15.00,
+                                                   1000.0 / 10.00,
+                                                   1000.0 / 5.000,
+                                                   1000.0 / 2.000,
+                                                   1000.0,
+                                                   2000.0};
+
+      const DT vblank_time = m_vps_counter.GetDtAvg() + m_vps_counter.GetDtStd();
+      const DT frame_time = m_fps_counter.GetDtAvg() + m_fps_counter.GetDtStd();
+      const double target_max_time = DT_ms(vblank_time + frame_time).count();
+      const double a =
+          std::max(0.0, 1.0 - std::exp(-4000.0 * m_vps_counter.GetLastRawDt().count() /
+                                       DT_ms(m_vps_counter.GetSampleWindow()).count()));
+
+      static double max_time = 0.0;
```

Comment:
note that this being mutably static makes the function inherently thread-unsafe to call, if that matters